### ISSUE TITLE
How to add and train new Entity? Clarifications

### ISSUE THREAD

**Issue by vinoth459** [explosion/spaCy#479]
Hi,

I have checked below link for adding new entity.
https://github.com/spacy-io/spaCy/issues/187

I need few clarifications from below example ( It has taken from above link ):

import plac

from spacy.en import English
from spacy.gold import GoldParse

def main(out_loc):

    nlp = English(parser=False) 

    doc = nlp.tokenizer(u'Lions and tigers and grizzly bears!')

    nlp.tagger(doc) 

    nlp.entity.add_label('ANIMAL') 

    indices = tuple(range(len(doc)))
    words = [w.text for w in doc]
    tags = [w.tag_ for w in doc]
    heads = [0 for _ in doc]
    deps = ['' for _ in doc]

    ner = ['U-ANIMAL', 'O', 'U-ANIMAL', 'O', 'B-ANIMAL', 'L-ANIMAL', 'O']

    annot = GoldParse(doc, (indices, words, tags, heads, deps, ner))

    loss = nlp.entity.train(doc, annot)
    i = 0
    while loss != 0 and i < 1000:
        loss = nlp.entity.train(doc, annot)
        i += 1
    print("Used %d iterations" % i)

    nlp.entity(doc) 
    // **what's happening here? why same line has given as input?**
    // **FYI I have tried with my own statement with ANIMAL content, it's not working**

    for ent in doc.ents:
        print(ent.text, ent.label_)
    nlp.entity.model.dump(out_loc) // **I able to store model, but how could I use it?**

if __name__ == '__main__':
    plac.call(main)
